project/npda/views/patient.py

```diff
@@ -162,32 +129,43 @@ def get_context_data(self, **kwargs):
         context["current_page"] = self.request.GET.get("page", 1)
         context["sort_by"] = self.get_sort_by()
 
+        # Add extra fields to the patient that we can't add to the query. This is ok because the queryset will be max the page size.
+        error_count_in_page = 0
+        valid_count_in_page = 0
+
+        for patient in context["page_obj"]:
+            # Signpost the latest quarter
+            if patient.most_recent_visit_date is not None:
+                patient.latest_quarter = retrieve_quarter_for_date(
+                    patient.most_recent_visit_date
+                )
+            
+            # Highlight the separation between patients with errors and those without
+            # unless we are sorting by a particular field in which case errors appear mixed
+            if not context["sort_by"]:
+                if not patient.is_valid or patient.visit_error_count > 0:
+                    if error_count_in_page == 0:
+                        patient.is_first_error = True
+                    
+                    error_count_in_page += 1
+
+                if patient.is_valid and patient.visit_error_count == 0:
+                    if valid_count_in_page == 0:
+                        patient.is_first_valid = True
+                    
+                    valid_count_in_page += 1
+
+        context["error_count_in_page"] = error_count_in_page
+        context["valid_count_in_page"] = valid_count_in_page
+
         return context
 
     def get(self, request, *args: str, **kwargs) -> HttpResponse:
         response = super().get(request, *args, **kwargs)
+        
         if request.htmx:
-            # filter the patients to only those in the same organisation as the user
```

Comment:
Everything still seems to work without this HTMX specific override. @eatyourpeas can you remember why we needed to have this special case?

Reply:
for sure could be removed. It guards against returning only the table partial template if it is an htmx request, rather than returning the whole page which is what would happen with a page reload. It is probable that it doesn't get called currently except from that htmx get request which i think only happens once the main page has loaded.